Source/Mothership/TMoth.cpp

```diff
@@ -310,11 +330,26 @@ unsigned TMoth::CmRun(string task)
     case TaskInfo_t::TASK_BARR:
         {
             DebugPrint("Task %s entering tinsel barrier\n",task.c_str());
-            P_Msg_Hdr_t barrier_msg;
-            barrier_msg.messageLenBytes = p_hdr_size(); // barrier is only a header. No payload.
-            barrier_msg.destEdgeIndex = 0;                     // no edge index necessary.
-            barrier_msg.destPin = P_SUP_PIN_INIT;              // it goes to the system __init__ pin
-            barrier_msg.messageTag = P_MSG_TAG_INIT;           // and is of message type __init__.
+            
+            //Assemble a Barrier message

```

Comment:
Can we abstract message assembly into a separate function? There are some commonalities between barrier-breaking and stop messages, for example

Reply:
Probably - the changes made to TMoth were very much minimal required to simplify and implement.